developer/src/tike/xml/layoutbuilder/cleanup.js

```diff
@@ -0,0 +1,79 @@
+$(function() {
+
+  builder.cleanupKVKL = function() {
+    let cleanupEmptyArrays = function() {
+      for (let platform in KVKL) {
+        for (let layer of KVKL[platform].layer) {
+          let newRow = [], n = 1;
+          for(let row of layer.row) {
+            if(row.key && Array.isArray(row.key)) {
+              for(let key of row.key) {
+                if(key.sk && (!Array.isArray(key.sk) || key.sk.length == 0)) {
+                  delete key.sk;
+                }
+              }
+              if(row.key.length > 0) {
+                row.id = n;
+                n++;
+                newRow.push(row);
+              }
+            }
+          }
+          layer.row = newRow;
+        }
+      }
+      return true;
+    };
+
+    let cleanupNumericField = function(obj, prop) {
```

Comment:
I don't see a case where this can return `false` - only `true`.  Granted, it is part of a boolean-logic condition chain in a different function, but it still strikes me a bit odd.